### PR TITLE
[GM-1608] github-integration.md: fix GitHub PR reporting with GitHub Actions

### PR DESCRIPTION
**Pull request by pcarrier**
Without it, the wrong commit ID is sent to Studio, which cannot match it with the PR.

### CODE REVIEW

**Comment by pcarrier** [comment]
Not sure why https://deploy-preview-1042--apollo-docs-index.netlify.app/docs/studio/github-integration/ 404s?

File: studio-docs/source/github-integration.md [hunk 1]
```diff
@@ -54,30 +55,36 @@ jobs:
...
-    
+pull_request:
+branches: [ main ]
+
 jobs:
-  schema-check:
-    # The type of runner that the job will run on any runner with node will work
-    runs-on: ubuntu-latest
+schema-check: # The type of runner that the job will run on any runner with node will work
+runs-on: ubuntu-latest
```
Comment (jsegaran) [inline studio-docs/source/github-integration.md:72]: Was the formatting change intentional?

File: studio-docs/source/github-integration.md [hunk 2]
```diff
@@ -54,30 +55,36 @@ jobs:
...
 
     # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
     - uses: actions/checkout@v2
+      with:
+        # Use the head of the PR instead of the merge commit
+        ref: ${{ github.event.pull_request.head.sha }}
```
Comment (jsegaran) [inline studio-docs/source/github-integration.md:79]: I think we do want to use the merge commit for schema reasons right? Specifically if I'm behind master I don't want to run a check on an old schema and get a spurious diff right?

File: studio-docs/source/github-integration.md (hunk 1 above)
Comment (pcarrier) [reply studio-docs/source/github-integration.md]: Nope, vscode did that.

File: studio-docs/source/github-integration.md (hunk 2 above)
Comment (pcarrier) [reply studio-docs/source/github-integration.md:79]: @jsegaran Are you suggesting monorepo is doing it wrong? :)

**Review by jsegaran — APPROVED** [review]

**Review by zionts — APPROVED** [review]
This looks great! I think we can document the "prospective merge" functionality in another section.

**Comment by pcarrier** [comment]
@jsegaran convinced me to go for the prospective merge primarily, left the head solution as a YAML comment.

**Review by zionts — APPROVED** [review]

**Pull request merged**